test/main_test.go

```diff
@@ -187,3 +191,54 @@ func showScenarios(features []*gherkin.Feature) {
 	}
 	mainLogger.Info("------------------ END SHOW SCENARIOS ------------------")
 }
+
+// This does just create a namespace, install the operators via subscription if crd is missing
+// and then delete the namespace
+// Should be deleted once https://issues.redhat.com/browse/KOGITO-1376 is solved
+func importDependentOperatorCrds() {
+	mainLogger := framework.GetMainLogger()
+	mainLogger.Infof("Install dependent operator crds if needed")
+	namespace := framework.GenerateNamespaceName("crd-install")
+
+	if err := framework.CreateNamespace(namespace); err != nil {
+		panic(err)
+	}
+
+	installErr := installDependentOperatorsIfNeeded(namespace)
+	if installErr != nil {
+		mainLogger.Errorf("Error installing dependent operators: %v", installErr)
+	}
+
+	if err := framework.DeleteNamespace(namespace); err != nil {
+		panic(err)
+	}
+
+	if installErr != nil {
+		panic(installErr)
+	}
+}
+
+// Should be deleted once https://issues.redhat.com/browse/KOGITO-1376 is solved
+func installDependentOperatorsIfNeeded(namespace string) error {
+	var operatorsToInstall []string
+	for operatorName := range framework.KogitoOperatorCommunityDependencies {
+		if installed, err := framework.IsCommunityOperatorCrdAvailable(operatorName); err != nil {
+			return err
+		} else if !installed {
+			operatorsToInstall = append(operatorsToInstall, operatorName)
+		}
+	}
+
+	for _, operatorName := range operatorsToInstall {
+		if err := framework.InstallCommunityKogitoOperatorDependency(namespace, operatorName); err != nil {
```

Comment:
So here we will install all dependent operators even if they are not needed? Because in this step:

https://github.com/kiegroup/kogito-cloud-operator/blob/01238f460145419857c7985f843bbf246081ad74/test/steps/operator.go#L55

We only install operators which are needed. If this PR gets merged, then that part of the step is not needed anymore I guess? Because the newly added code will always make sure that all operators (thus all CRDs) are installed before running the scenario.

Reply:
We just want the crds to be installed as a workaround.
So we don't even wait for them to be running.
when deleting the namespace, they will be automatically deleted.
But for the Kogito operator to be running, we need all dependent operators' crds installed. If not, the Kogito operator will not be running, even if we don't need dependent operators...
That is why this quick fix. Will take less than 2 min but will avoid all tests to fail because of missing crds.

Reply:
in the future, KOGITO-1376 should solve this problem in Kogito Operator

Reply:
ah, forgot about the different namespace, sorry :)